language/move-model/src/model.rs

```diff
@@ -2398,6 +2398,11 @@ impl<'env> FunctionEnv<'env> {
             || match self.definition_view().visibility() {
                 Visibility::Public => true,
                 Visibility::Script => panic!("Script visibility not yet supported"),
+                // TODO: for simplicity, we may treat `public(friend)` the same way as `public`
+                // based on the fact that they both expose the function to external modules. We may
+                // want to change the function name from `is_public` to `is_exposed` or have
+                // iner-grained return values (instead of just a bool) to indicate visibility.
+                Visibility::Friend => panic!("Friend visibility not yet supported"),
```

Comment:
nit, I might have done this for the script case... but you can do `unimplemented!(..)` which gives a *slightly* better message when it is hit